// mypackage/chart.js
function capitalizeFirstLetter(string) {
    return string.charAt(0).toUpperCase() + string.slice(1).toLowerCase();
  }
  
  function calculateAge(birthDate) {
    const today = new Date();
    const birthDateObj = new Date(birthDate);
    let age = today.getFullYear() - birthDateObj.getFullYear();
    const monthDiff = today.getMonth() - birthDateObj.getMonth();
    
    if (monthDiff < 0 || (monthDiff === 0 && today.getDate() < birthDateObj.getDate())) {
      age--;
    }
    
    return age.toString();
  }
  
  function transformObject(inputObj) {
    const id = Math.random().toString(36).substr(2, 10);
    const firstName = capitalizeFirstLetter(inputObj.name.split(' ')[0]);
    const lastName = capitalizeFirstLetter(inputObj.name.split(' ')[1]);
    const age = calculateAge(inputObj.dateBirth);
    const purpose = capitalizeFirstLetter(inputObj.purpose);
    
    const outputObj = {
      id: id,
      firstName: firstName,
      lastName: lastName,
      dateBirth: inputObj.dateBirth,
      age: age,
      purpose: purpose
    };
    
    return outputObj;
  }
  
  module.exports = { transformObject }
