// mypackage/main.js
    let data = {
        name: '',
        dateBirth: '',
        purpose: '',
          };

const id = Math.random().toString(36).substring(2, 12);
const birthday = new Date('10.03.1987');
const dataBirth = birthday.toLocaleDateString();
const words = data.name.split(' ');
const firstName = words[0][0];
// const firstName = words[0].charAt(0).toUpperCase() + words[0].substring(1);

// const lastName =  words[1].charAt(0).toUpperCase() + words[1].substring(1);

const age = Math.floor ((new Date() - birthday) / 31556952000 );
const purpose = () => {
  let purpose =  data.purpose[0].toUpperCase() + data.purpose.substring(1);
  return purpose;
  console.log('purpose: ', purpose);
} 

class Person {
    constructor() {
        this.id = id;
        this.firstName = firstName; 
        // this.lastName =lastName;
        this.dataBirth = dataBirth;
        this.age = age;
        this.purpose = purpose;
        }

    //  let person = new Person();

// }
}
module.exports = { Person, firstName,  purpose };
